circus-train-gcp/src/test/java/com/hotels/bdp/circustrain/gcp/context/GCPBeanPostProcessorTest.java

```diff
@@ -16,46 +16,78 @@
+import com.hotels.bdp.circustrain.gcp.FileSystemFactory;
+import com.hotels.bdp.circustrain.gcp.GCPCredentialCopier;
+import com.hotels.bdp.circustrain.gcp.GCPCredentialPathProvider;
+import com.hotels.bdp.circustrain.gcp.RandomStringFactory;
 
 @RunWith(MockitoJUnitRunner.class)
 public class GCPBeanPostProcessorTest {
-  private @Spy GCPSecurity security = new GCPSecurity();
-  private @Spy Configuration configuration = new Configuration();
+  private @Mock RandomStringFactory randomStringFactory;
+
+  private @Mock GCPSecurity security;
+  private @Mock GCPCredentialPathProvider credentialPathProvider;
+  private @Mock DistributedFileSystemPathProvider distributedFileSystemPathProvider;
+  private @Mock Configuration configuration;
+  private @Mock BindGoogleHadoopFileSystem bindGoogleHadoopFileSystem;
+  private @Mock FileSystemFactory fileSystemFactory;
+  private @Mock GCPCredentialCopier credentialCopier;
+
+  private GCPBeanPostProcessor processor;
+
+  @Before
+  public void init() {
+    doNothing().when(configuration).set(anyString(), anyString());
+    processor = new GCPBeanPostProcessor(credentialPathProvider, distributedFileSystemPathProvider,
+        bindGoogleHadoopFileSystem, fileSystemFactory, credentialCopier);
+  }
 
   @Test
   public void postProcessAfterInitializationWithCorrectBeanName() throws Exception {
     String beanName = CommonBeans.BEAN_BASE_CONF;
-    GCPBeanPostProcessor processor = new GCPBeanPostProcessor(security);
     processor.postProcessAfterInitialization(configuration, beanName);
-    verify(security, times(1)).getCredentialProvider();
+    verify(credentialPathProvider).newPath();
   }
 
   @Test
   public void postProcessAfterInitializationWithIncorrectBeanName() throws Exception {
     String beanName = "notBaseConf";
-    GCPBeanPostProcessor processor = new GCPBeanPostProcessor(security);
     processor.postProcessAfterInitialization(configuration, beanName);
-    verify(security, times(0)).getCredentialProvider();
+    verify(credentialPathProvider, times(0)).newPath();
   }
 
   @Test
   public void postProcessAfterInitializationWithBlankCredentialProviderDoesntModifyConfiguration() throws Exception {
     String beanName = CommonBeans.BEAN_BASE_CONF;
-    GCPBeanPostProcessor processor = new GCPBeanPostProcessor(security);
     processor.postProcessAfterInitialization(configuration, beanName);
     when(security.getCredentialProvider()).thenReturn("");
-    verify(security, times(1)).getCredentialProvider();
+    verify(credentialPathProvider).newPath();
     verify(configuration, times(0)).set(anyString(), anyString());
   }
-}
\ No newline at end of file
+
+  @Test
+  public void postProcessorWithNonNullCredentialProvider() {
+    String beanName = CommonBeans.BEAN_BASE_CONF;
+    doReturn(new Path("/test.json")).when(credentialPathProvider).newPath();
+    processor.postProcessAfterInitialization(configuration, beanName);
+    verify(credentialPathProvider).newPath();
```

Comment:
In general: if you have a `when(credentialPathProvider).newPath()` you don't need a `verify(credentialPathProvider).newPath()` as the verify will be superfluous.